gradle-baseline-java-config/resources/checkstyle/checkstyle.xml

```diff
@@ -123,11 +123,17 @@
             <message key="import.illegal" value="Use JUnit 4-style (org.junit.*) test classes and assertions instead of JUnit 3 (junit.framework.*)."/>
         </module>
         <module name="IllegalImport"> <!-- Java Coding Guidelines: Import the canonical package -->
-            <property name="illegalPkgs" value="org.elasticsearch.common.base, jersey.repackaged.com.google.common, com.google.api.client.repackaged, org.apache.hadoop.thirdparty.guava, com.clearspring.analytics.util, org.spark_project.guava"/>
+            <property name="illegalPkgs" value="org.elasticsearch.common.base, com.clearspring.analytics.util, org.spark_project.guava"/>
+            <message key="import.illegal" value="Must not import repackaged classes."/>
+        </module>
+        <module name="IllegalImport"> <!-- Java Coding Guidelines: Import the canonical package -->
+            <property name="illegalPkgs" value=".*.(repackaged|shaded|thirdparty)"/>
```

Comment:
is this a prefix, or do we need a suffix too?

Reply:
We don't need a suffix

Reply:
Should the second dot be escaped?

Reply:
Would be ideal if we had tests for this.

Reply:
Escaping it actually breaks the rule.

Reply:
Is the “.*.” component even necessary then? (ie is this actually a regex?)

Reply:
As per checkstyle docs, it is treated as a regex when we set regexp to true. From testing it out, the `.*.` component does seem necessary.

Reply:
Seems like I was mistaken about the dot escaping. It works without escaping the dot, but escaping it doesn't break the rule.

Reply:
I think it needs to be escaped or these will also match “xrepackaged”